### PR TITLE
add pass_credentials to helm source

### LINKED ISSUE #269: Add support for `pass_credentials` in helm
### Description

Helm sometimes requires the `pass_credentials` flag to be set on the `argocd_application`.  The provider should support this.

### Potential Terraform Configuration

```hcl
  spec {
    destination {
      server = "https://kubernetes.default.svc"
    }

    source {
      helm {
        release_name = local.service_name
        values       = local.helm_values
        pass_credentials = true
      }
    }
  }
```

### References

https://argo-cd.readthedocs.io/en/stable/user-guide/helm/#helm-pass-credentials

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### PR DESCRIPTION
**Pull request by duncan485**
closes  #269

### CODE REVIEW

**Review by onematchfox — CHANGES_REQUESTED** [review]
Hi @duncan485,

Thanks for this! The changes look great. 

To answer your [comment](https://github.com/oboukili/terraform-provider-argocd/issues/269#issuecomment-1533564588) - no schema updates are needed in this case, so you're all good. 

One small ask from my side... Could you please add this new field (`pass_credentials = true`) to the existing test for Helm charts [here](https://github.com/duncan485/terraform-provider-argocd/blob/59297183d2589e49550d959629b88fdc232b58da/argocd/resource_argocd_application_test.go#L1102-L1149)?

File: argocd/resource_argocd_application_test.go [hunk 1]
```diff
@@ -1127,7 +1127,7 @@ resource "argocd_application" "helm" {
           value = "standalone"
         }
 
-	pass_credentials = true
+        pass_credentials = true
```
Comment (duncan485) [inline argocd/resource_argocd_application_test.go:1130]: Was already a bit confused, thanks for the quick fix 😄

**Review by onematchfox — APPROVED** [review]
LGTM. Thanks for taking the time to do this! Will merge once tests are passing

**Comment by onematchfox** [comment]
@duncan485, thanks again! This will be released along with the next version - currently targeting early next week, pending review on #268/#273 as well as my availability due to some upcoming travel. If you need it released sooner, then let me know, and I'll do my best to help out.

**Pull request merged**